paste/src/views/ProfileComponent.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useMemo, useState } from 'react';
+import { PasteProps } from 'paste/pages/[id]';
+import styled from '@emotion/styled';
+
+const CLOSE_ICON =
+    'url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA10lEQVR4Xt2Tu8rCQBSEzzlJmYBvYJ7EShArfQ1BrPNDyoB5DRHsrUQQQc3lcfxBiKbIHjfiCuayKSwEB6ZYPpjdGVhkZvhEJP3dALMMgmB+FoI7ddUQEYhw7bp/48YAkYtOfzCEJu22m5G2gnjeHIYxxHGiXJwV11dQTzdN4w0QKa4JULplGcymkwo4Rkn7iCjNOcNiuVK3vQZ0ug5gWwAgPnpalgW1+yDqA4hQmmSADVVGBW8bES7RaW+zYOBSNSSENL0etAGe5/UkMKBZ/77vv8APfKY7cvZVTt7VqzwAAAAASUVORK5CYII=)';
+const OPEN_ICON =
+    'url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA7ElEQVR4Xt1TzYrCQAxOYo8z0GeRZY+eBPGkryGIZ4UeC/Yx/EHvnqTssot/9XF2YUHXQyd2RtuC2OnBg2AgJF/CfJkvwyAzwyNGiT+XwLktBMHwRyl270lDRCDCRb8/aBcSqFi59UYTiuwzXLasEtR18nYbQRTt4f2tqqPGad8uIb2641TyKUSJp307gbH/0wl63U4Ks3y925e/AibOMcN4OofRZKZLOmps6lj2CoBodAohspKUMtePaCcgQqNZCGlw+PFt8nwXWLZE+NttviQrBr6RhoRwOBxXVgLP82pJqECx/fq+n4EX+ExnBI9csQQ1hIoAAAAASUVORK5CYII=)';
+
+interface ProfileEntry {
+    name: string;
+    selfTime: number;
+    children: ProfileEntry[];
+    depth: number;
+    parent?: ProfileEntry;
+}
+
+const INVALID_PROFILE = {
+    name: 'Invalid Profile',
+    selfTime: 0,
+    depth: 0,
+    children: []
+};
+
+function parseLine(line: string): { name: string; selfTime: number } {
```

Comment:
This function is pretty far away from where it's used. Move down towards `ProfileComponent`?